File: 2020/go/day15/main_test.go
package main

import (
	"testing"

	"github.com/peter554/adventofcode/2020/go/lib"
)

func TestPart1(t *testing.T) {
	sample := []int{0, 3, 6}
	lib.Expect(t, 436, Part1(sample))

	input := []int{1, 0, 16, 5, 17, 4}
	lib.Expect(t, 1294, Part1(input))
}

func TestPart2(t *testing.T) {
	sample := []int{0, 3, 6}
	lib.Expect(t, 175594, Part2(sample))

	input := []int{1, 0, 16, 5, 17, 4}
	lib.Expect(t, 573522, Part2(input))
}

File: 2024/ruby/day05/main_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative 'main'

class TestMain < Minitest::Test
  def test_part1
    assert_equal 143, part1('day05/example')
    assert_equal 5964, part1('day05/input')
  end

  def test_part2
    assert_equal 123, part2('day05/example')
    assert_equal 4719, part2('day05/input')
  end
end
